provider-example/src/hpke.rs

```diff
@@ -213,14 +211,28 @@ impl HpkeOpener for HpkeRsReceiver {
     }
 }
 
+use error_util::other_err;
+
 #[cfg(feature = "std")]
-fn other_err(err: impl StdError + Send + Sync + 'static) -> Error {
-    Error::Other(OtherError(Arc::new(err)))
+mod error_util {
+    use alloc::sync::Arc;
+    use rustls::Error;
+    use rustls::OtherError;
+    use std::error::Error as StdError;
+
+    pub(crate) fn other_err(err: impl StdError + Send + Sync + 'static) -> Error {
+        Error::Other(OtherError(Arc::new(err)))
+    }
 }
 
 #[cfg(not(feature = "std"))]
-fn other_err(err: impl Send + Sync + 'static) -> Error {
-    Error::General(alloc::format!("{}", err));
+mod error_util {
+    use core::any::Any;
+    use rustls::Error;
```

Comment:
oops forgot this, apologies

```suggestion
    use rustls::Error;
    use rustls::OtherError;
```